modules/apps/portal-workflow/portal-workflow-test/src/testIntegration/java/com/liferay/portal/workflow/service/test/WorkflowDefinitionLinkServiceImplTest.java

```diff
@@ -178,6 +178,57 @@ public void testGetWorkflowDefinitionLinks() throws Exception {
 			workflowDefinitionLink2.getWorkflowDefinitionName());
 	}
 
+	@Test
+	public void testUpdateWorkflowDefinitionLink() throws Exception {
+		_setUpPermissionThreadLocal(_companyAdminUser);
+
+		ConfigurationTestUtil.saveConfiguration(
+			_configuration,
+			HashMapDictionaryBuilder.<String, Object>put(
+				"company.administrator.can.publish", true
+			).build());
+
+		WorkflowDefinitionLink workflowDefinitionLink1 =
+			_addWorkflowDefinitionLink(
+				"Single Approver", BlogsEntry.class.getName());
+
+		User user = _addUser();
+
+		_setUpPermissionThreadLocal(user);
+
+		AssertUtils.assertFailure(
+			PrincipalException.MustBeCompanyAdmin.class,
+			StringBundler.concat(
+				"User ", user.getUserId(), " must be the company ",
+				"administrator to perform the action"),
+			() -> _workflowDefinitionLinkService.updateWorkflowDefinitionLink(
+				workflowDefinitionLink1.getExternalReferenceCode(),
+				TestPropsValues.getUserId(), TestPropsValues.getCompanyId(),
+				workflowDefinitionLink1.getGroupId(),
+				"com.liferay.account.model.AccountEntry", 0, 0,
+				"Single Approver",
+				workflowDefinitionLink1.getWorkflowDefinitionVersion()));
+
+		_setUpPermissionThreadLocal(_companyAdminUser);
+
+		WorkflowDefinitionLink workflowDefinitionLink2 =
+			_workflowDefinitionLinkService.updateWorkflowDefinitionLink(
```

Comment:
Do you think we should also cover the scenario by keeping the class name and changing the workflowDefinitionName?